### ISSUE TITLE
Compaction blocked.

### ISSUE THREAD

**Issue by LucienXian** [facebook/rocksdb#10384]
When we write a DB with high concurrency, we find that L0 accumulates a lot of files, which leads to the occurrence of write stall or even write stop. Through the relevant logs and codes, we found that there will be multiple jobs that merge L0 files concurrently, instead of L0->L1 compaction([hit this](https://github.com/facebook/rocksdb/blob/main/db/compaction/compaction_picker_level.cc#L558)), because the flush speed is faster than the speed of L0 merge, which leads to L0's sst files accumulate quickly.

### Expected behavior
Faster L0->L1 compaction, write stops or write stalls are resolved eventually.  
Or is it possible to turn off the L0->L0 behavior, we do not need to provide high-performance random reads, and can accept that there are multiple files in the L0 layer at certain times.

### Actual behavior
Since there is always a merge between L0 (L0->L0), the compaction of L0->L1 is not performed. Even if the files in the L0 layer have far exceeded level0_file_num_compaction_trigger.

### Steps to reproduce the behavior
Write some data  with high concurrency.

**Comment by mdcallag** [comment]
I have yet to look closely at compaction scheduling in practice, with respect to the "hit this" URL you share above.

Intra-L0 can't be disabled today although that is easy to add if you are wiling to modify upstream source.

When https://github.com/facebook/rocksdb/pull/10057 lands in a release (hopefully 7.5) then worst-case write stalls are less of an issue.

**Comment by rockeet** [comment]
@LucienXian  Our in house branch([link]()) have a [fix](https://github.com/topling/toplingdb/commit/76da3562468f954d772bb36b9362f11655a39922) for this issue: if set `level0_file_num_compaction_trigger  <= 0`, Intra-L0 compaction will be disabled.

**Comment by LucienXian** [comment]
> I have yet to look closely at compaction scheduling in practice, with respect to the "hit this" URL you share above.
> 
> Intra-L0 can't be disabled today although that is easy to add if you are wiling to modify upstream source.
> 
> When #10057 lands in a release (hopefully 7.5) then worst-case write stalls are less of an issue.

Thx. I took a quick look at the PR you mentioned. For my situation, can I think that after this PR is completed, it will effectively improve the situation of multiple concurrent intra-L0s blocking L0->Lbase compaction, which is to directly disable Intra -L0 yet? In addition, what does the worst-case write stalls mentioned here refer to?

**Comment by rockeet** [comment]
> > level0_file_num_compaction_trigger
> 
> @rockeet Does simply setting level0_file_num_compaction_trigger affect the L0->Lbase compaction decision behavior?

Yes, just level0_file_num_compaction_trigger effect Intra-L0 compaction, with my [fix](https://github.com/topling/toplingdb/commit/76da3562468f954d772bb36b9362f11655a39922), Intra-L0 compaction can be disabled by setting level0_file_num_compaction_trigger <= 0.
